src/utils.js

```diff
@@ -4,52 +4,67 @@ import validateFormData from "./validate";
 
 const widgetMap = {
   boolean: {
-    checkbox: "CheckboxWidget",
-    radio: "RadioWidget",
-    select: "SelectWidget",
-    hidden: "HiddenWidget",
+    default: "CheckboxWidget",
+    all: {
+      checkbox: "CheckboxWidget",
+      radio: "RadioWidget",
+      select: "SelectWidget",
+      hidden: "HiddenWidget",
+    },
   },
   string: {
-    text: "TextWidget",
-    password: "PasswordWidget",
-    email: "EmailWidget",
-    hostname: "TextWidget",
-    ipv4: "TextWidget",
-    ipv6: "TextWidget",
-    uri: "URLWidget",
-    "data-url": "FileWidget",
-    radio: "RadioWidget",
-    select: "SelectWidget",
-    textarea: "TextareaWidget",
-    hidden: "HiddenWidget",
-    date: "DateWidget",
-    datetime: "DateTimeWidget",
-    "date-time": "DateTimeWidget",
-    "alt-date": "AltDateWidget",
-    "alt-datetime": "AltDateTimeWidget",
-    color: "ColorWidget",
-    file: "FileWidget",
+    default: "TextWidget",
+    all: {
+      text: "TextWidget",
+      password: "PasswordWidget",
+      email: "EmailWidget",
+      hostname: "TextWidget",
+      ipv4: "TextWidget",
+      ipv6: "TextWidget",
+      uri: "URLWidget",
+      "data-url": "FileWidget",
+      radio: "RadioWidget",
+      select: "SelectWidget",
+      textarea: "TextareaWidget",
+      hidden: "HiddenWidget",
+      date: "DateWidget",
+      datetime: "DateTimeWidget",
+      "date-time": "DateTimeWidget",
+      "alt-date": "AltDateWidget",
+      "alt-datetime": "AltDateTimeWidget",
+      color: "ColorWidget",
+      file: "FileWidget",
+    },
   },
   number: {
-    text: "TextWidget",
-    select: "SelectWidget",
-    updown: "UpDownWidget",
-    range: "RangeWidget",
-    radio: "RadioWidget",
-    hidden: "HiddenWidget",
+    default: "TextWidget",
+    all: {
+      text: "TextWidget",
+      select: "SelectWidget",
+      updown: "UpDownWidget",
+      range: "RangeWidget",
+      radio: "RadioWidget",
+      hidden: "HiddenWidget",
+    },
   },
   integer: {
-    text: "TextWidget",
-    select: "SelectWidget",
-    updown: "UpDownWidget",
-    range: "RangeWidget",
-    radio: "RadioWidget",
-    hidden: "HiddenWidget",
+    default: "TextWidget",
```

Comment:
I feel like `integer` default widget should be `UpDownWidget` but that's a breaking change, we can do that next major release.